sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -495,7 +531,7 @@ private void acquireChannel(final Promise<Channel> promise) {
                 return;
             }
 
-            final int channelCount = this.channels();
+            final int channelCount = this.channels(false);
```

Comment:
Much of troubleshooting/investigation context is fresh in minds but will have a high decay factor. 
Thoughts on capturing that fully context in the code.

Reply:
Added comment